src/realm-roles/AssociatedRolesModal.tsx

```diff
@@ -143,19 +145,22 @@ export const AssociatedRolesModal = (props: AssociatedRolesModalProps) => {
     refresh();
   }, [filterType]);
 
-  useFetch(
-    async () => {
-      if (id) return await adminClient.roles.findOneById({ id });
-    },
-    (fetchedRole) => {
-      if (fetchedRole) {
-        setName(fetchedRole.name!);
-      } else {
-        setName(t("createRole"));
-      }
-    },
-    []
-  );
+  {
+    !props.isMapperId &&
+      useFetch(
+        async () => {
+          if (id) return await adminClient.roles.findOneById({ id });
+        },
+        (fetchedRole) => {
+          if (fetchedRole) {
+            setName(fetchedRole.name!);
+          } else {
+            setName(t("createRole"));
+          }
+        },
+        []
+      );
+  }
```

Comment:
It's [not allowed](https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level) to render hooks conditionally, so this has to be a single `useFetch` with any conditional logic inside the function body.
```suggestion
  useFetch(
    () =>
      props.isMapperId
        ? adminClient.roles.findOneById({ id })
        : Promise.resolve(null),
    (role) => setName(role ? role.name! : t("createRole")),
    []
  );
```